pkg/RepoManages.go

```diff
@@ -387,15 +403,71 @@ func (impl RepoManagerImpl) FetchChanges(pipelineMaterialId int, from string, to
 		return response, nil
 	}
 	commits := make([]*git.GitCommit, 0)
-	err = json.Unmarshal([]byte(pipelineMaterial.CommitHistory), &commits)
+	err := json.Unmarshal([]byte(pipelineMaterial.CommitHistory), &commits)
 	if err != nil {
 		return nil, err
 	}
 	response.Commits = commits
 	return response, nil
+}
+
+
+func (impl RepoManagerImpl) FetchGitCommitsForPrType(pipelineMaterial *sql.CiPipelineMaterial) (*git.MaterialChangeResp, error) {
+	response := &git.MaterialChangeResp{}
 
+	pipelineMaterialId := pipelineMaterial.Id
+
+	webhookMappings, err := impl.webhookEventRepository.GetCiPipelineMaterialPrWebhookMapping(pipelineMaterialId)
+	if err != nil {
+		impl.logger.Errorw("error in getting webhook mapping for pipelineId ", "id", pipelineMaterialId, "errMsg", err)
+		return nil, err
+	}
+
+	if len(webhookMappings) == 0 {
+		impl.logger.Infow("no webhook mapping for ci pipeline, pipelineId", "pipelineId", pipelineMaterialId)
+		return nil, err
+	}
+
+	var prWebhookEventIds []int
+	for _, webhookMapping := range webhookMappings {
+		prWebhookEventIds = append(prWebhookEventIds, webhookMapping.PrWebhookDataId)
+	}
+
+	prWebhookEvents, err := impl.webhookEventRepository.GetOpenPrEventDataByIds(prWebhookEventIds, 15)
+	if err != nil {
+		impl.logger.Errorw("error in getting webhook for ids ", "ids", prWebhookEventIds, "errMsg", err)
+		return nil, err
```

Comment:
err will be nil in this case, is that the intention? then we should send nil explicity

Reply:
done